utils/micro-controllers.stanza

```diff
@@ -71,6 +71,21 @@ public defn micro-controller (params:Tuple<KeyValue>) -> (Tuple<KeyValue<Symbol,
   val component = to-jitx(mcu)
   mcu-module(component)
 
+public defn find-micro-controller (core-parameters:Tuple<KeyValue<String, ?>>, 
+                            module-parameters:Tuple<KeyValue<Symbol, ?>>,  
+                            mcu-design) :
+  val possible-mcus = MicroControllers(core-parameters, 10)
```

Comment:
Why 10? Is that just a way to limit it to the 10 best? What if none of the 10 works in the pin search?

Reply:
Then it fails. This design tries the 10 smallest given the parameters. Checking each one is quite slow. I could make the number to check an agrument if you want.

Reply:
adding as a parameter defaulted to 10 is a good strategy

Reply:
The default limit is 25 across the board
```
public defn dbquery-all (params: Tuple<KeyValue<String, Tuple<String>|Tuple<Double>|String|Double|Int>>) -> Tuple<JSON> :
  dbquery(params, 25)
```
However I haven't defined constructors without limit argument fro Resistors, Inductors, Capacitors, MicroControllers. But better not start mixing different values for default limits